framework/src/main/java/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -546,15 +577,35 @@ private AnnotationMirror createAnnotationForStringConcatenation(
         List<String> leftValues = getStringValues(leftOperand, p);
         List<String> rightValues = getStringValues(rightOperand, p);
 
+        boolean allowNullStringConcat =
```

Comment:
This isn't allowing `null`, it's assuming that the operand of a string concatenation might be null.